swig/cpp/src/Session.h

```diff
@@ -50,11 +51,11 @@ class Session:public Iter_Value, public Values
     void list_schemas(Schema& schema);
     void get_schema(Schema& schema, const char *module_name, const char *revision,
```

Comment:
I am working on this, the only problem is that SWIG does not tolerate shared_ptr<char> and the caller has to free the memory so I had to create a new class (Schema_Content) for this.

With the shared_ptr functionality I will be able to simplify some functions:
`void get_last_error(Errors& err);` -> `shared_ptr<Errors> get_last_error();`

callbacks will now return classes instead sysrepo C structs.

I wanted to ask you is it better to have:
`shared_ptr<Schema_Content> get_schema(const char *module_name, const char *revision, const char *submodule_name, sr_schema_format_t format);`
or
`shared_ptr<Schema_Content> get_schema(const char *module_name, sr_schema_format_t format, const char *revision = NULL, const char *submodule_name = NULL);`

With the second API in python we would have:
`schema = sess.get_schema("ietf-interfaces", sr.SR_SCHEMA_YANG)`
instead
`schema = sess.get_schema("ietf-interfaces", None, None, sr.SR_SCHEMA_YANG)`

I will be done with the refactoring tomorrow, so the shared_ptr will be used where it is possible.
The other part I wanted to mention is that I will probably wrap the C structs in sysrepo.h to C++ objects because the C struct wrapping is inconsistent among binding languages in SWIG.